.ci.sh

```diff
@@ -39,16 +39,13 @@ run_tests() {
 }
 
 run_tests_with_coverage(){
-  # NB: Coverage does not currently work: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/2702
+  # NB: Testing using example from https://github.com/codecov/example-python
   activate_venv_sct
-  pip install coverage
-  echo -ne "import coverage\ncov = coverage.process_startup()\n" > sitecustomize.py
-  echo -ne "[run]\nconcurrency = multiprocessing\nparallel = True\n" > .coveragerc
-  COVERAGE_PROCESS_START="$PWD/.coveragerc" COVERAGE_FILE="$PWD/.coverage" pytest
-  coverage combine
+  pytest --cov=spinalcordtoolbox --cov-config setup.cfg --cov-branch --cov-report=xml:cov-api.xml testing/api
+  pytest --cov=spinalcordtoolbox --cov-config setup.cfg --cov-branch --cov-report=xml:cov-cli.xml testing/cli
```

Comment:
Re: the context given in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3463#discussion_r786162517, it might be a bit tricky to get coverage, since `pytest` is only used to validate the end results of the pipeline. In other words, `pytest` isn't used to invoke the processing itself, so I'm not sure if we can hook into any line-by-line coverage scanning. :(